### PR TITLE
Force MSVC version

### PR DESCRIPTION
**Pull request by groutr**
This allows a recipe to force which version of MSVC it builds with.

### CODE REVIEW

**Comment by ccordoba12** [comment]
Thanks @groutr and sorry for not creating this PR before :-) Got busy with other stuff.

I have a couple of comments:

1. I'd like to pass any Visual Studio version to conda-build. But if I understand things correctly, your PR would only let me override between `msvc10` and `msvc14`.
2. You need to set

        DISTUTILS_USE_SDK=1
        MSSdk=1

    in the environment, to override `distutils` Visual Studio detection and use the compiler you want for Python extensions.

File: conda_build/windows.py [hunk 1]
```diff
@@ -61,19 +61,18 @@ def msvc_env_cmd():
...
         else:
-            vcvarsall = os.path.join(program_files,
-                                     r'Microsoft Visual Studio 10.0'
-                                     r'\VC\vcvarsall.bat')
+            vcvarsall = vcvarsall.format(version='10.0')
     else:
-        vcvarsall = os.path.join(program_files,
-                                 r'Microsoft Visual Studio 9.0'
-                                 r'\VC\vcvarsall.bat')
+        vcvarsall = vcvarsall.format(version='9.0')
```
Comment (ccordoba12) [inline conda_build/windows.py]: According to my comments, this block could be rewritten as:

```python
vs_versions = {'msvc9': '9.0', 'msvc10': '10.0', 'msvc11': '11.0',
               'msvc12': '12.0', 'msvc14': '14.0'}
if override:
    vcvarsall = vcvarsall.format(version=vs_versions[override])
    os.environ['DISTUTILS_USE_SDK'] = '1'
    os.environ['MSSdk'] = '1'
elif config.PY3K:
    if config.use_MSVC2015:
        vcvarsall = vcvarsall.format(version='14.0')
    else:
        vcvarsall = vcvarsall.format(version='10.0')
else:
    vcvarsall = vcvarsall.format(version='9.0')
```

**Comment by ccordoba12** [comment]
Pinging @msarahan and @asmeurer. Those who are working on Windows packages really need this functionality :-)

File: conda_build/windows.py (hunk 1 above)
Comment (msarahan) [inline conda_build/windows.py]: +1, this is cleaner.

**Comment by msarahan** [comment]
+1 - nice work.  Please make edits according to @ccordoba12 suggestions, but otherwise ready to go.

**Comment by groutr** [comment]
@ccordoba12, changing it so you set msvc_compiler like `msvc_compiler: 10.0`

**Comment by groutr** [comment]
I will test the latest changes in my windows vm either tonight or tomorrow.

**Comment by ccordoba12** [comment]
+1 from me. It seems ok :-)

**Comment by groutr** [comment]
merging.

**Comment by asmeurer** [comment]
Can you make a pull request to conda-docs that adds this to the docs. And you should note that this is only for advanced users who know what they are doing, since for most people, changing the compiler from the default (at least for Python extensions) will break things.

**Pull request merged**